zoom/utils.go

```diff
@@ -31,19 +38,29 @@ func connect(ctx context.Context, d *plugin.QueryData) (*zoom.Client, error) {
 	if zoomConfig.APISecret != nil {
 		apiSecret = *zoomConfig.APISecret
 	}
+	if zoomConfig.AccountID != nil {
+		accountID = *zoomConfig.AccountID
+	}
+	if zoomConfig.ClientID != nil {
+		clientID = *zoomConfig.ClientID
+	}
+	if zoomConfig.ClientSecret != nil {
+		clientSecret = *zoomConfig.ClientSecret
+	}
 
-	if apiKey == "" || apiSecret == "" {
+	if (accountID == "" || clientID == "" || clientSecret == "") && (apiKey == "" || apiSecret == "") {
 		// Credentials not set
-		return nil, errors.New("api_key and api_secret must be configured")
+		return nil, errors.New("Server-to-Server oauth app or JWT app credentials must be configured")
 	}
 
-	// Configure to automatically wait 1 sec between requests, per Zoom API requirements
-	conn := zoom.NewClient(apiKey, apiSecret)
-
-	// Save to cache
-	d.ConnectionManager.Cache.Set(cacheKey, conn)
-
-	return conn, nil
+	// prefer server-to-server oauth app creds
+	if accountID != "" || clientID != "" || clientSecret != "" {
```

Comment:
```suggestion
	if accountID != "" && clientID != "" && clientSecret != "" {
```

I think this logic is safer, as we want to check if all 3 args are defined before trying to create an OAuth client

Reply:
yes, updated.